### DISCUSSION TITLE
eslint plugin works even if there's no directly import

### DISCUSSION THREAD

**Discussion by Jungzl** [TanStack/query#4638] [Ideas]
Personally, I'm using `unplugin-auto-import` plugin and has config like below
```typescript
AutoImport({
    imports: [
      'vue',
      'vue-router',
      { '@tanstack/vue-query': ['useQuery', 'useQueries', 'useInfiniteQuery', 'useMutation', 'useQueryClient'] },
    ],
  })
```
so that I can use functions in `@tanstack/vue-query` directly. 
I knew that this plugin only works when it detects directly import. 
Can we make it works ? e.g. an option guarantees that there are no functions with the same name in the project

**Comment by TkDodo** [comment]
not sure. @Newbie012 ? Not having false positives is more important imo

**Reply by Newbie012** [reply]
We could a config property called `skipImportCheck`

**Reply by Jungzl** [reply]
Yes, this gonna works

**Reply by TkDodo** [reply]
as long as it defaults to `false` :)
